### ISSUE TITLE
Cannot pass a11y validation for complex Mantine components

### ISSUE THREAD

**Issue by lichwala** [mantinedev/mantine#8125]
### Dependencies check up

- [x] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

8.2.2

### What package has an issue?

@mantine/dates @mantine/notifications etc. with complex components

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

Some Mantine complex components (like DateInput displaying a callendar) have some a11y issues with buttons - no aria-label is applied. In the latest Mantine 8.2.1 we finally have a possibility to attach custom attributes for the **internal parts** of the component:
```
<DateInput
  attributes={{
    calendarHeaderControl: { foo: 'bar' }
  }}></DateInput>
```
which properly generates attribute `foo="bar"` for **both** *previous/next* callendar buttons, BUT:

### Problem 1:
There is no possibility to apply `aria-label` attribute in that way. This code does not work:
```
<DateInput
  attributes={{
    calendarHeaderControl: { 'aria-label': 'bar' }
  }}></DateInput>
```
No attribute is generated for code above. Note: I don't want to attach `aria-label` for the whole `DateInput` component, but for its internal *previous/next* buttons.

### Problem 2:
There is no possibility to apply different `aria-label` values for *previous* and *next* buttons, because we cannot use `data-*` selector in attributes. So this code does not work:
```
<DateInput
  attributes={{
    'calendarHeaderControl[data-direction="previous"]': { 'aria-label': 'Previous month' },
    'calendarHeaderControl[data-direction="next"]': { 'aria-label': 'Next month' },
  }}></DateInput>
```

So I think it would be nice to have some mechanism to distinguish both such buttons (while attaching attributes to Mantine components), as we can do this in CSS:

```
  .mantine-DateInput-calendarHeaderControl {
    &[data-direction="previous"] {
      color: blue;
    }

    &[data-direction="next"] {
      color: red;
    }
  }
```

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
You can customize these with `nextLabel` and `previousLabel` props.
You can also use `ariaLabels` prop to assign various aria labels, see type definition to learn more.

**Comment by lichwala** [comment]
In case of `@mantine/notifications` extension there is a problem with *Close* button (visible by default) for a single notification popup - it does not pass a11y validation because it has no `aria-label` attached, but currently there is no possibility in Mantine to attach such attribute to this button.

**Comment by rtivital** [comment]
Use `closeButtonProps={{ 'aria-label': 'Close' }}` to assign aria label to notification. Please use component props tables to learn about available features.

**Comment by lichwala** [comment]
Both issues solved - thank you very much!

> Please use component props tables to learn about available features.

I usually do, but did't find it somehow even fighting with this for a few hours... :-) Sorry for bothering.

Thank you. Mantine is awesome!
